src/components/shared/ContractFunction.vue

```diff
@@ -0,0 +1,366 @@
+<script setup lang="ts">
+import useWeb3 from '@/compositions/useWeb3'
+import { BN } from 'bn.js'
+import { reactive, computed, watch, onMounted } from 'vue'
+import { toWei, Unit } from 'web3-utils'
+import { MethodSelect, MethodType } from '../endpoints/SendTransaction.vue'
+import ParamField from './ParamField.vue'
+
+interface ElementType extends MethodType {
+  error?: boolean
+}
+type Props = {
+  call?: string
+  modelValue?: MethodType[]
+  custom?: boolean
+  onlyParams?: boolean
+  hideData?: boolean
+  data?: string
+  testidPrefix?: string
+}
+
+type Emits = {
+  (event: 'update:modelValue', value: any): void
+  (event: 'update:error', value: boolean): void
+  (event: 'update:data', value: string): void
+}
+
+const props = defineProps<Props>()
+const emits = defineEmits<Emits>()
+
+const { getWeb3 } = useWeb3()
+const { eth } = getWeb3()
+
+const SIGNATURE_CACHE = 'signature-cache'
+
+type BytesSignatureResponse = {
+  count: number
+  next: unknown
+  previous: unknown
+  results: [
+    {
+      id: number
+      created_at: string
+      text_signature: string
+      hex_signature: string
+      bytes_signature: string
+    }
+  ]
+}
+
+const fetcher = async <Response, Request>(config: {
+  url: string
+  method: 'GET' | 'POST'
+  data?: Request
+  headers?: Record<string, never>
+}): Promise<Response> => {
+  const fetchConfig: RequestInit = {
+    method: config.method,
+    headers: config.headers || {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+  }
+  if (config.data) {
+    fetchConfig.body = JSON.stringify(config.data)
+  }
+
+  const response = await fetch(config.url, fetchConfig)
+
+  if (!response.ok) {
+    return response
+      .json()
+      .catch(() => {
+        throw new Error(response.status.toString())
+      })
+      .then(message => {
+        throw message
+      })
+  }
+  return await response.json()
+}
+
+const decodeData = async (data: string): Promise<MethodSelect> => {
+  if (/^0x/i.test(data)) {
+    data = data.substring(2)
+  }
+  const selector = data?.substring(0, 8)
+  if (selector) {
+    const signatureCache = await caches.open(SIGNATURE_CACHE)
+    const url = `https://www.4byte.directory/api/v1/signatures/?hex_signature=${selector}`
```

Comment:
Please move url to `config.ts`

Reply:
Moving.

Reply:
I like the idea of grouping by standard, but that causes a lot of repeats. I tried to group it by function and some of the functions are available on multiple standards. So although it says Mint LSP8 the underlying function is mint with different arguments than mint on ERC20 for example. It would be possible to use two dropdowns on with Standards ERC20,ERC777,... and another one with Mint, Transfer, Send and so on. Might add a lot of complications though.

Reply:
Changed the order to this. Might be more user-friendly.
![image](https://user-images.githubusercontent.com/708186/210236344-def8293d-28f5-4364-9490-99e39163c250.png)
I think a two menu approach would sort it better, but adding a lot of additional code and then the custom items won't be sorted anyways.